coaddAnalysis.py

```diff
@@ -0,0 +1,234 @@
+                          fontsize=fontSize - 1, horizontalalignment="center", verticalalignment="center")
+    axes.set_xlabel('RA', size='large')
+    axes.set_ylabel('Dec', size='large')
+    axes.set_xlim(xlim)
+    axes.set_ylim(ylim)
+
+
+def mosaicCoadd(butler, patch_list, band='g', tract=0, ref_patch=None, sampling=100, norm=None, nImage=False,
+                fig=None, show_colorbar=True, filename=None):
+    """Generate a mosaic image of many coadded patches.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.base.Butler`
+        A Butler instance to retrieve data from a repository.
+    patch_list : `list` of `str`
+        A list of the patches containing images to mosaic.
+    band : `str`, optional
+        The band of the coadd to retrieve from the repository.
+    tract : `int`, optional
+        The tract of the skyMap.
+    ref_patch : `str`, optional
+        If set, use the given patch to set the image normalization for the figure.
+    sampling : `int`, optional
+        Stride factor to sample each input image in order to reduce the size in memory.
+        A `sampling` of 1 will attempt to display every pixel.
+    norm : `astropy.visualization.ImageNormalize`, optional
+        Normalization to set the color scale of the images.
+    nImage : `bool`, optional
+        Mosaic the nImage instead of the coadd.
+    fig : `matplotlib.pyplot.fig`, optional
+        Figure instance to display the mosaic in.
+    show_colorbar : `bool`, optional
+        Display a colorbar on the figure.
+    filename : `str`, optional
+        If set, write the figure to a file with the given filename.
+    """
+    coaddName = 'deepCoadd'
+    if nImage:
+        coaddName += '_nImage'
+    if norm is None:
+        if ref_patch in patch_list:
+            dataId = {'filter': band, 'tract': tract, 'patch': ref_patch}
+            coadd = (butler.get(coaddName, dataId=dataId)).getImage().getArray()
+            norm = ImageNormalize(coadd, interval=ZScaleInterval(), stretch=SqrtStretch())
+    patch_x = []
+    patch_y = []
+    for patch in patch_list:
+        m = patch.find(",")
+        patch_x.append(int(patch[:m]))
+        patch_y.append(int(patch[m+1:]))
+    x0 = min(patch_x)
+    y0 = min(patch_y)
+    nx = max(patch_x) - x0 + 1
+    ny = max(patch_y) - y0 + 1
+
+    if fig is None:
+        fig = plt.figure(figsize=(ny, nx))
+    gs1 = gridspec.GridSpec(ny, nx)
+    gs1.update(wspace=0.0, hspace=0.0)  # set the spacing between axes.
+    for x in range(nx):
+        for y in range(ny):
+            i = x + nx*y
+            ax = plt.subplot(gs1[i])
+            patch = f"{x + x0},{ny - y + y0}"
+            if patch in patch_list:
+                dataId = {'filter': band, 'tract': tract, 'patch': patch}
+                coadd = (butler.get(coaddName, dataId=dataId)).getImage().getArray()
+                coadd = np.flip(coadd[::sampling, ::sampling], axis=0)
```

Comment:
I understand the `sampling` part, but why is is flipped?

Reply:
I should make this an option. For this data, it needs to be flipped to match the layout of the patches.